demo/src/main/java/com/example/demo/repo/VideoRepository.java

```diff
@@ -0,0 +1,43 @@
+package com.example.demo.repo;
+
+import com.example.demo.models.UserDataBase;
+import com.example.demo.models.Video;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public interface VideoRepository extends JpaRepository<Video, Long> {
+    @Transactional
+    public Video findVideoByName(String nameVideo);
+
+    @Transactional
+    public Video findVideoById(Long id);
+
+
+    @Transactional
+    public Video findVideoByUserDataBaseAndId(UserDataBase user, Long id);
+
+    @Transactional
+    public Video findVideoByIdentifier(String identifier);
+
+    @Transactional
+    public List<Video> findAll();
+
+    @Transactional
+    @Query(value = "UPDATE videos SET position = subquery.position " +
```

Comment:
No way. Such big updates are very rarely used. And never used if there is any other way.
Here "other way" is to sort by position (subquery.position but it won't be a subquery - just a query) and return first video.